packages/web-app-files/src/mixins/actions/virusScan.ts

```diff
@@ -0,0 +1,54 @@
+import { clientService } from 'web-pkg/src/services'
+import { isLocationSpacesActive } from '../../router'
+
+export default {
+  computed: {
+    $_virus_scan_items() {
+      return [
+        {
+          name: 'virus-scan',
+          icon: 'virus',
+          label: () => this.$gettext('Scan for viruses'),
+          handler: this.$_virus_scan_trigger,
+          isEnabled: ({ resources }) => {
+            if (resources.length === 0) {
+              return false
+            }
+
+            if (resources[0].isFolder) {
+              return false
+            }
+
+            if (
+              !isLocationSpacesActive(this.$router, 'files-spaces-personal') &&
+              !isLocationSpacesActive(this.$router, 'files-spaces-project')
+            ) {
+              return false
+            }
+
+            if (
+              isLocationSpacesActive(this.$router, 'files-spaces-share') &&
+              resources[0].path === '/'
+            ) {
+              return false
+            }
+
+            return true
+          },
+          componentType: 'button',
+          class: 'oc-files-actions-virus-scan-trigger'
+        }
+      ]
+    }
+  },
+  methods: {
+    async $_virus_scan_trigger({ resources }) {
+      const accessToken = this.$store.getters['runtime/auth/accessToken']
+      const httpClient = clientService.httpAuthenticated(accessToken)
+      const { server } = this.$store.getters.configuration
```

Comment:
Actually we prefer using the (relatively new) `ConfigurationManager` for this. You can retrieve the URL via `configurationManager.serverUrl` then.

Reply:
i keep this for now, ill add this later... release is pressing :)